streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -122,7 +126,15 @@ public static FileLock lockStateDirectory(File stateDir) throws IOException {
 
     private static FileLock lockStateDirectory(File stateDir, int retry) throws IOException {
         File lockFile = new File(stateDir, ProcessorStateManager.LOCK_FILE_NAME);
-        FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
+        FileChannel channel = null;
+        synchronized (channels) {
+            channel = channels.get(lockFile);
+            if (channel == null) {
+                channel = new RandomAccessFile(lockFile, "rw").getChannel();
+                channels.put(lockFile, channel);
```

Comment:
This scenario can happen when a rebalance migrates some tasks from one thread to another, within the same process / JVM.

And I think we should still try to close the channel while shutting down the processor state manager, since otherwise the number of channels will just grow indefinitely due to rebalances that move tasks around, although very slowly.